torch/overrides.py

```diff
@@ -1739,7 +1739,12 @@ def is_tensor_like(inp):
 def _wrap_torch_function(f):
     @functools.wraps(f)
     def wrapped(self, *args, **kwargs):
-        with enable_torch_function_mode(self.inner):
+        if hasattr(self, "inner"):
+            inner = self.inner
+        else:
+            inner = None
+
+        with enable_torch_function_mode(inner):
```

Comment:
I remember we were debugging something related to inner being BaseMode or something last week, and that this PR should fix it. If so, then that would be worth adding as a test case